projects/packages/account-protection/src/class-password-detection.php

```diff
@@ -79,32 +100,44 @@ public function login_form_password_detection( $user, string $password ) {
 	public function handle_password_detection_validation_error( string $username, \WP_Error $error ): void {
 		if ( isset( $error->errors['password_detection_validation_error'] ) ) {
 			$token = $error->get_error_data()['token'];
-			wp_safe_redirect( $this->get_redirect_url( $token ) );
-			exit;
+			$this->redirect_and_exit( $this->get_redirect_url( $token ) );
 		}
 	}
 
+	/**
+	 * Load user by ID. Dependency decoupling.
+	 *
+	 * @param int $user_id The user ID.
+	 *
+	 * @return \WP_User|null The user object.
+	 */
+	protected function load_user( int $user_id ) {
+		return get_user_by( 'ID', $user_id );
```

Comment:
I tried hard to find a way to solve this with filters or something, but was ultimately unable to do so.

For example, I tried using `wp_cache_set` so that this function could just read the user from the cache under the hood, but that didn't work with WordDBless.

I even added `Brain\Monkey` as a dev requirement, however, then found that it does not seem to go well together with WorDBless, as that is loaded first and then the function already exists, so brain monkey can not redefine it.

Happy to hear suggestions for better solutions.